File: PYTHON/SivitasAkademik.py
# import kelsa human karena sivitas akademik merupakan anak dari kelas human
from Human import Human
# membuat kelas sivitias akademik
class SivitasAKademik(Human):
    # kelas ini memiliki 2 atribut private
    __asal_univ = ""
    __email_univ =""
    
    # konstruktor yang paramternya berisi atribut orangtuanya dan atribut aslinya 
    def __init__(self, nik="", nama="", jenis_kelamin="", asal_univ = "", email_univ = ""):
        # supet untuk menjalankan init kelas orangtuanya    
        super().__init__(nik, nama, jenis_kelamin)
        self.__asal_univ = asal_univ
        self.__email_univ = email_univ
        
    # getter setter
    # get asal_univ
    def get_asal_univ(self):
        return self.__asal_univ
    # set asal_univ
    def set_asal_univ(self, asal_univ):
        self.__asal_univ = asal_univ
    # get email_univ
    def get_email_univ(self):
        return self.__email_univ
    # set email_univ
    def set_email_univ(self, email_univ):
        self.__email_univ = email_univ
        
    

File: CPP/SivitasAkademik.cpp

#include "header.hh"

SivitasAkademik::SivitasAkademik() // konstruktor kosong
    : Human() // artinya sivitas akademik inheritance human
{
    this->asal_univ = "";
    this->email_edu = "";
}
SivitasAkademik::SivitasAkademik(string nik, string nama, string jenis_kelamin, string asal_univ, string email_edu) // kosntruktor parameter
    : Human(nik, nama, jenis_kelamin) // artinya sivitas akademik inheritance human dan menjalankan kosntruktor human
{
    this->asal_univ = asal_univ;
    this->email_edu = email_edu;
}

// SETTER & GETTER
string SivitasAkademik::get_asal_univ()
{
    return asal_univ;
}
void SivitasAkademik::set_asal_univ(string asal_univ)
{
    this->asal_univ = asal_univ;
}
string SivitasAkademik::get_email_edu()
{
    return email_edu;
}
void SivitasAkademik::set_email_edu(string email_edu)
{
    this->email_edu = email_edu;
}

SivitasAkademik::~SivitasAkademik() // destruktor
{

}